dev-requirements.txt

```diff
@@ -14,7 +14,7 @@ sphinx_autodoc_typehints
 matplotlib
 scipy
 pre-commit
-typing-extensions; python_version < '3.8'
```

Comment:
Before removing it, do you have an idea why this was enforced? Maybe @erikmannerfelt would know.

Reply:
My guess is to support typing for Python < 3.7 which was probably not enabled by default in typing-extensions, as typing was not so widespread then. Should be good now.